File: Programmers/L2_프린터.py
from collections import deque

def solution(priorities, location):
    answer = 0
    queue = deque([(i, p) for i, p in enumerate(priorities)])
    priorities.sort()
    
    while queue:
        i, q = queue.popleft()
        p = priorities.pop()
        
        if p > q:
            queue.append((i, q))
            priorities.append(p)
        else:
            answer += 1
            if i == location:
                return answer

File: Programmers/L2_프린터.js
function solution(priorities, location) {
  let answer = 0;
  let queue = priorities.map((p, i) => { return [i, p]});
  priorities.sort((a, b) => a - b);

  while (queue) {
    let [i, q] = queue.shift();
    p = priorities.pop();

    if (p > q) {
      queue.push([i, q]);
      priorities.push(p);
    } else {
      answer += 1
      if (i == location) {
        return answer;
      };
    };
  };
}
